eradiate/scenes/biosphere.py

```diff
@@ -46,7 +48,46 @@ class BiosphereFactory(BaseFactory):
 
 @attr.s
 class Canopy(SceneElement, ABC):
-    """An abstract base class defining a base type for all canopies."""
+    """
+    An abstract base class defining a base type for all canopies.
+    """
+
+    id = attr.ib(
+        default="canopy",
+        validator=attr.validators.optional(attr.validators.instance_of(str)),
+    )
+
+    size = attrib_quantity(
+        default=None,
+        # TODO: add other validators maybe
+        validator=validator_has_len(3),
+        units_compatible=cdu.generator("length"),
+    )
+
+    padding = attr.ib(
```

Comment:
Do you still need this attribute, given that you have transferred padding handling to the scene level?